keras/backend/cntk_backend.py

```diff
@@ -2065,20 +2065,20 @@ def stop_gradient(variables):
 
 
 def switch(condition, then_expression, else_expression):
-    ndim_cond = ndim(condition)
-    ndim_expr = ndim(then_expression)
-    if ndim_cond > ndim_expr:
+    cond_ndim = ndim(condition)
+    expr_ndim = ndim(then_expression)
+    if cond_ndim > expr_ndim:
         raise ValueError('Rank of condition should be less'
                          ' than or equal to rank of then and'
                          ' else expressions. ndim(condition)=' +
                          str(cond_ndim) + ', ndim(then_expression)'
                          '=' + str(expr_ndim))
```

Comment:
`cond_ndim` and `expr_ndim` was undefined.
I chose to use `expr_ndim` instead `ndim_expr` to be consistent with tensorflow_backend.py and theano_backend.py

Reply:
Please add a test checking that the ValueError is being raised correctly (`pytest.raises(ValueError)`)